src/mlpack/methods/cf/cf_main.cpp

```diff
@@ -142,17 +161,139 @@ void ComputeRecommendations(CFModel* cf,
 
     Log::Info << "Generating recommendations for " << users.n_elem << " users."
         << endl;
-    cf->GetRecommendations(numRecs, recommendations, users.row(0).t());
+
+    //  Making All the possible paths
+    if (ns_algo == "Cosine")
+    {
+        if (iw_algo == "Average")
```

Comment:
We use two spaces for one tab : https://github.com/mlpack/mlpack/wiki/DesignGuidelines#tabbing.